pkg/cli/image/imagesource/options.go

```diff
@@ -35,8 +45,48 @@ func (o *Options) Repository(ctx context.Context, ref TypedImageReference) (dist
 		url := ref.Ref.DockerClientDefaults().RegistryURL()
 		return driver.Repository(ctx, url, ref.Ref.RepositoryName(), o.Insecure)
 	default:
-		return nil, fmt.Errorf("unrecognized image reference type %s", ref.Type)
+		return nil, nil, fmt.Errorf("unrecognized image reference type %s", ref.Type)
+	}
+}
+
+// PreferredImageSource returns an appropriate image reference, repository, and manifest service  for a given Context, or an error. Alternative image sources
```

Comment:
Since we already run Context.Repository from imagesource.Options, I added the new PreferredImageSource method here, as well.